app/controllers/publishers/ats_api/v1/vacancies_controller.rb

```diff
@@ -30,11 +30,11 @@ def update
 
     result = Publishers::AtsApi::UpdateVacancyService.call(vacancy, permitted_vacancy_params)
 
-    if result[:success]
+    if result[:status] == :ok
       @vacancy = vacancy
       render :show
     else
-      render json: { errors: result[:errors] }, status: :unprocessable_entity
+      render result.slice(:json, :status)
```

Comment:
by making the update return the same interface as the create, resolved the issues with some errors not being captured/returned as expected.